plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,581 @@
+        checked_func = function()
+            return G_reader_settings:isTrue("press_key_does_hotkeyshortcuts")
+        end,
+        callback = function()
+            G_reader_settings:flipNilOrFalse("press_key_does_hotkeyshortcuts")
+            UIManager:askForRestart()
+        end,
+    }
+    menu_items.hotkeyshortcuts = {
+        sorting_hint = "physical_buttons_setup",
+        text = _("Keyboard shortcuts"),
+        sub_item_table = {
+            {
+                text = _("Cursor keys"),
+                sub_item_table = self:genSubItemTable(cursor_keys),
+            },
+            {
+                text = _("Page-turn buttons"),
+                enabled_func = function() return self.hotkey_mode == "hotkeyshortcuts_reader" end,
+                sub_item_table = self:genSubItemTable(pg_turn),
+            },
+            {
+                text = _("Function keys"),
+                sub_item_table = self:genSubItemTable(fn_keys),
+            },
+        },
+    }
+
+    if Device:hasKeyboard() then
+        table.insert(menu_items.hotkeyshortcuts.sub_item_table, {
+            text = _("Alphabet keys"),
+            sub_item_table = self:genSubItemTable({
+                "alt_plus_a", "alt_plus_b", "alt_plus_c", "alt_plus_d", "alt_plus_e", "alt_plus_f", "alt_plus_g", "alt_plus_h", "alt_plus_i",
+                "alt_plus_j", "alt_plus_k", "alt_plus_l", "alt_plus_m", "alt_plus_n", "alt_plus_o", "alt_plus_p", "alt_plus_q", "alt_plus_r",
+                "alt_plus_s", "alt_plus_t", "alt_plus_u", "alt_plus_v", "alt_plus_w", "alt_plus_x", "alt_plus_y", "alt_plus_z",
+            }),
+        })
+    end
+end
+
+--[[
+    Description:
+    This function overrides existing `registerKeyEvents()` functions in various modules to resolve conflicts and customize key event handling.
+    It modifies the key event registration for several reader and file manager modules based on device capabilities and user settings.
+
+    Modules and their modifications:
+    - ReaderBookmark: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderConfig: Customizes `ShowConfigMenu` key event based on user settings.
+    - ReaderDictionary: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderLink: Customizes `GotoSelectedPageLink` key event for devices with screen keyboard or symbol key.
+    - ReaderSearch: Customizes `ShowFulltextSearchInputBlank` key event for devices with a keyboard.
+    - ReaderToc: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderThumbnail: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderWikipedia: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderUI: Customizes `Home` and `KeyContentSelection` key events based on device capabilities.
+    - FileManager: Customizes `Home` and `Back` key events, and conditionally removes `Close` key event.
+    - FileSearcher: CCustomizes `ShowFileSearchBlank` key event for devices with keyboards.
```

Comment:
(Double `CC` in `CCustomizes`.)

Looking at the code below, your `Customizes...` feels like `Overrides... with a non-empty function`.
When reading these comments first, I thought you would extend (ie. call the original first) rather than overrides.

Reply:
they are conscious-uncoupling customised overrides... :)

> When reading these comments first, I thought you would extend (ie. call the original first) rather than overrides.

is that better?

Reply:
Technically no, you indeed do need to override them all I guess.
I'm just talking about the comments: using "Customizes" is more vague than "Override" (which you do) that you use anyway for the others.

Reply:
well I guess I am using customise as a form of soft override, and override as a full override

Reply:
You're using "customise" as "override with new stuff".
And "override" as "override with a no-op".

Reply:
> You're using "customise" as "override with new stuff".

not true, I am not really adding anything (the exception being ReaderConfig) to them. I just keep the necessary bits that already exists in the original functions. and just remove the stuff that either uses `ScreenKB` or `Shift`

Reply:
Yes, but technically, you're overriding them with something similar. You're overriding everything, with either a no-op or a stripped clone of the original code. You could call that customize (I wouldn't). But ok, I was just mentionning it, not minding, it's your plugin after all :)

Reply:
You're the native English writer, so you'd fine something more easily than me.
But it's fine, I was just mentionning it.